src/Avalonia.Controls/Primitives/ScrollBar.cs

```diff
@@ -506,5 +513,211 @@ private void UpdatePseudoClasses(Orientation o)
+                var thumbLength = scrollBar.Orientation == Orientation.Vertical ? track.Thumb?.Bounds.Height ?? 0 : track.Thumb?.Bounds.Width ?? 0;
+                var clickPosition = scrollBar.Orientation == Orientation.Vertical ? scrollBar._lastRightClickPosition.Y : scrollBar._lastRightClickPosition.X;
+                
+                if (trackLength > thumbLength)
+                {
+                    var ratio = clickPosition / trackLength;
+                    var range = scrollBar.Maximum - scrollBar.Minimum;
+                    var value = scrollBar.Minimum + (ratio * range);
+                    scrollBar.SetCurrentValue(ValueProperty, Math.Max(scrollBar.Minimum, Math.Min(scrollBar.Maximum, value)));
+                    scrollBar.OnScroll(ScrollEventType.ThumbTrack);
+                }
+            }
+        }
+
+        private static void ScrollToTopAction(ScrollBar scrollBar)
+        {
+            scrollBar.SetCurrentValue(ValueProperty, scrollBar.Minimum);
+            scrollBar.OnScroll(ScrollEventType.SmallDecrement);
+        }
+
+        private static void ScrollToBottomAction(ScrollBar scrollBar)
+        {
+            scrollBar.SetCurrentValue(ValueProperty, scrollBar.Maximum);
+            scrollBar.OnScroll(ScrollEventType.SmallIncrement);
+        }
+
+        private static void ScrollToLeftEndAction(ScrollBar scrollBar)
+        {
+            scrollBar.SetCurrentValue(ValueProperty, scrollBar.Minimum);
+            scrollBar.OnScroll(ScrollEventType.SmallDecrement);
+        }
+
+        private static void ScrollToRightEndAction(ScrollBar scrollBar)
+        {
+            scrollBar.SetCurrentValue(ValueProperty, scrollBar.Maximum);
+            scrollBar.OnScroll(ScrollEventType.SmallIncrement);
+        }
+
+        private static void PageUpAction(ScrollBar scrollBar)
+        {
+            scrollBar.LargeDecrement();
+        }
+
+        private static void PageDownAction(ScrollBar scrollBar)
+        {
+            scrollBar.LargeIncrement();
+        }
+
+        private static void PageLeftAction(ScrollBar scrollBar)
+        {
+            scrollBar.LargeDecrement();
+        }
+
+        private static void PageRightAction(ScrollBar scrollBar)
+        {
+            scrollBar.LargeIncrement();
+        }
+
+        private static void LineUpAction(ScrollBar scrollBar)
+        {
+            scrollBar.SmallDecrement();
+        }
+
+        private static void LineDownAction(ScrollBar scrollBar)
+        {
+            scrollBar.SmallIncrement();
+        }
+
+        private static void LineLeftAction(ScrollBar scrollBar)
+        {
+            scrollBar.SmallDecrement();
+        }
+
+        private static void LineRightAction(ScrollBar scrollBar)
+        {
+            scrollBar.SmallIncrement();
+        }
+
+        private Track? Track => this.GetTemplateChildren().OfType<Track>().FirstOrDefault();
+
+        private class SimpleCommand : ICommand
```

Comment:
We shouldn't really put classes in the same file, even private ones. This could be used other places in the future. On top of that -- why not just use a more standard RelayCommand-based approach?